diff --git a/cppapi/include/tango/client/ApiUtil.h b/cppapi/include/tango/client/ApiUtil.h
index 938ef7a2..5d136e9c 100644
--- a/cppapi/include/tango/client/ApiUtil.h
+++ b/cppapi/include/tango/client/ApiUtil.h
@@ -236,7 +236,7 @@ private:
     };
 
 	TANGO_IMP static ApiUtil 	*_instance;
-	static omni_mutex			inst_mutex;
+	TANGO_IMP static omni_mutex			inst_mutex;
 	bool						exit_lock_installed;
 	bool						reset_already_executed_flag;
 
