diff --git a/configure/check_omniidl.cmake b/configure/check_omniidl.cmake
index 2f1c17d0..25865990 100644
--- a/configure/check_omniidl.cmake
+++ b/configure/check_omniidl.cmake
@@ -17,8 +17,8 @@ function(test_omniidl)
 
     if (WIN32)
         # Always statically link on Windows so we don't need the omniorb dll's to be in the PATH
-        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreadedDebug")
-        set(link_libs ${OMNIORB_PKG_LIBRARIES_STA_DEBUG};ws2_32.lib)
+        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreadedDLL")
+        set(link_libs ${OMNIORB_PKG_LIBRARIES_DYN_RELEASE};ws2_32.lib)
         set(defs "-D__x86__" "-D__NT__" "-D__OSVERSION__=4" "-D__WIN32__" "-D_WIN32_WINNT=0x0400")
         foreach(def ${static_defs})
             list(APPEND defs "-D${def}")
